### ISSUE TITLE
Error after authorize connection to discord

### ISSUE THREAD

**Issue by Wazted** [Bearer/Pizzly#181]
Hi,

I'm currently having an error after all authentications i do with Pizzly.

`Error: Authorization cancelled. The user has likely interrupted the process by closing the modal.`

Here is my code : 
```
connect: function () {
            this.$pizzly.integration('instagram').connect().then(this.connectSuccess).catch(this.connectError);
        },
        connectytb: function () {
            this.$pizzly.integration('youtube').connect().then(this.connectSuccess).catch(this.connectError);
        },
        connectdisco: function () {
            this.$pizzly.integration('discord').connect().then(this.connectSuccess).catch(this.connectError);
        },
        connectSuccess: function (data) {
            // On success, we update the user object
            console.log(data);
            // this.user = data.authId;
            // this.fetchStarringRepositories();
        },
        connectError: function (err) {
            console.error(err);
            alert('Something went wrong. Look at the logs.');
        }
```

and the template : 
```
<template>
    <div>
        <vue-navigation-bar :options="navbarOptions" @vnb-item-clicked="logout" />
        <div class="modal">
            <b-modal id="user" title="User manage">
                <div class="d-block text-center">
                    <v-btn
                        tile
                        style="border-radius: 5px; font-size: 18px; font-family: 'IBM Plex Sans'"
                        height="64px"
                        width="360px"
                        outlined
                        x-large
                        color="white"
                        @click="handleClickSignIn"
                    >
                        Sign in with
                        <img id="logo" :src="this.googleIcon" />
                    </v-btn>
                    <v-btn
                        tile
                        style="border-radius: 5px; font-size: 18px; font-family: 'IBM Plex Sans'; margin-top: 20px"
                        height="64px"
                        width="360px"
                        outlined
                        x-large
                        color="white"
                        @click="connect"
                    >
                        Sign ion with
                        <img id="logo" :src="this.googleIcon" />
                    </v-btn>
                    <v-btn
                        tile
                        style="border-radius: 5px; font-size: 18px; font-family: 'IBM Plex Sans'; margin-top: 20px"
                        height="64px"
                        width="360px"
                        outlined
                        x-large
                        color="white"
                        @click="connectytb"
                    >
                        Sign issn with
                        <img id="logo" :src="this.googleIcon" />
                    </v-btn>
                    <v-btn
                        tile
                        style="border-radius: 5px; font-size: 18px; font-family: 'IBM Plex Sans'; margin-top: 20px"
                        height="64px"
                        width="360px"
                        outlined
                        x-large
                        color="white"
                        @click="connectdisco"
                    >
                        Sign injn with
                        <img id="logo" :src="this.googleIcon" />
                    </v-btn>
                </div>
            </b-modal>
        </div>
    </div>
</template>
```

I configured pizzly like that : 
```
import Pizzly from 'pizzly-js';

//mounted is in the script of sure
    mounted: function () {
        // Here we initialize Pizzly.
        this.$pizzly = new Pizzly({
            host: 'dashpizzly.herokuapp.com'
        });

        // If you don't have an account on Bearer.sh
        // create one for free at https://www.bearer.sh/signup
        // then grab your Bearer Publishable key
        // (pk_production...) in the settings page.
    },
```

**Comment by markmichon** [comment]
Hey @Wazted, I'm not up to date on Vue, but I suspect it might be the way you're handling everything inside `connect`. Have you seen [this article on vue and pizzly](https://dev.to/bearer/how-to-use-an-oauth-based-api-in-vue-js-1elo)? If not, it might help narrow down the problem.

**Comment by ddveloper** [comment]
I am seeing same error message, I simply downloaded the [React example code](https://codesandbox.io/s/pizzly-github-react-demo-rq78z) and run `npm install` then `npm start` locally on my PC, after click on the button, chrome console log this error: ```Error: Authorization cancelled. The user has likely interrupted the process by closing the modal.```

**Comment by markmichon** [comment]
Thanks @ddveloper, was able to reproduce. Looks to be a problem with the js client after 0.2.5. Will investigate, but in the meantime you can use `v0.2.5` as a workaround.

**Comment by ddveloper** [comment]
> Thanks @ddveloper, was able to reproduce. Looks to be a problem with the js client after 0.2.5. Will investigate, but in the meantime you can use `v0.2.5` as a workaround.

using 0.2.5 I am seeing CORS error instead:
```Access to fetch at 'xxx.herokuapp.com/proxy/github/user?pizzly_pkey=yyyy' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.```

**Comment by markmichon** [comment]
For those affected, if you haven't already (particularly with the CORS issue), make sure you're using the latest version of Pizzly along with the latest version of the pizzly-js client. Using a slightly modified version of the example @ddveloper shared, here's a [codesandbox that is working for me](https://codesandbox.io/s/pizzly-github-react-example-1ofbc). Make sure to sub in your pizzly instance and publishable key, as well as have github configured (or swap out the integration for your preferred API).

**Comment by Slluxx** [comment]
Hey, im also getting this error.
i downloaded pizzly-js via node to integrate it into my vuejs application. Has this not beeing fixed yet? which version can i implement safely without worrying about it and cors?

**Comment by markmichon** [comment]
Hey @Slluxx, I'm unable to reproduce your problem. Make sure that pizzly itself is working on whatever platform you have it deployed (like Heroku), and then provide more information if you find you're still having the problem. For now, I'm closing this issue as it is otherwise resolved when using the default settings.

**Comment by Slluxx** [comment]
I did not include the Pizzly via script tag because it would throw an exception that Pizzly is not defined when opening the site. Instead i used npm install to install the latest version to my project and imported Pizzly like this:
(BTW: I think this should be the default and not including it via JS tag like in your example)

`import Pizzly from '../../node_modules/pizzly-js/dist/index.umd'`

```
mounted: function () {
    this.$pizzly = new Pizzly({ host: 'myinsecurepizzlyinstance.herokuapp.com' })
  },
  created: function () {
  },
  methods: {
    connect: function () {
      this.$pizzly
        .integration('discord')
        .connect()
        .then(({ authId }) => {
          console.log(authId)
        })
        .catch((error) => {
          console.error(error)
        })
    }
  }
  ```
  
  I am calling `this.connect` via a button click event. 
The newest version throws an error but reverting to **0.2.5 fixes it with no code changes**.

**Comment by Slluxx** [comment]
@markmichon v0.2.5 works and i am getting an authId but when trying to use that ID like this:

```js
this.pizzly = new Pizzly({ host: 'embedmanager.herokuapp.com' }).integration('discord')
this.pizzly.auth(this.authId).get('/users/@me')
        .then((response) => console.log(response))
        .catch(console.error)
```

The answer im getting is a CORS error:

`Access to fetch at 'http://embedmanager.herokuapp.com/proxy/discord/users/@me' from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request.`

This seems expected behavior as discussed above but also means my integration in itself works. So whats up with the latest version installed via NPM by me?

**Comment by Slluxx** [comment]
I am very sorry @markmichon but this issue is not resolved and i am getting pretty frustrated.
I have set up a new vue project and a new heroku instance just to be sure. As said above 0.2.5 works fine except CORS error when trying to actually reach API/Proxy. 0.2.8 doesn't even let me log in.

**NPM version:**
7.5.3

**Node version:**
v15.10.0

**Error message:**
`Error: Authorization cancelled. The user has likely interrupted the process by closing the modal.`

**package.json:**
```json
{
  "name": "pizzlytest",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "serve": "vue-cli-service serve",
    "build": "vue-cli-service build"
  },
  "dependencies": {
    "core-js": "^3.6.5",
    "pizzly-js": "^0.2.8",
    "vue": "^2.6.11"
  },
  "devDependencies": {
    "@vue/cli-plugin-babel": "~4.5.0",
    "@vue/cli-service": "~4.5.0",
    "vue-template-compiler": "^2.6.11"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not dead"
  ]
}

```

**page.vue**
```vue
<template>
  <div class="hello">
    <button @click="connect()"> {{this.test}} </button>
    <p v-if="repositories">{{this.repositories}}</p>
  </div>
</template>
<script>
import Pizzly from 'pizzly-js'
export default {
  name: 'HelloWorld',
  data: function () {
    return {
      test: "connect",
      user: null,
      repositories: null
    }
  },
  created: function () {
    this.$pizzly = new Pizzly({
      host: "mypizzlyapp.herokuapp.com"
    });
  },
  methods: {
    connect: function() {
      this.$pizzly
        .integration('discord')
        .connect()
        .then(this.connectSuccess)
        .catch(this.connectError);
    },
    connectSuccess: function(data) {
      console.log(data)
      this.user = data.authId;
      this.fetchStarringRepositories()
    },
    connectError: function (err) {
      console.error(err)
      alert("Something went wrong. Look at the logs.")
    },
    fetchStarringRepositories: function() {
      this.$pizzly
        .integration("discord")
        .auth(this.user)
        .get("/users/@me")
        .then(response => response.json())
        .then((data) => { this.repositories = JSON.stringify(data) })
        .catch(this.fetchError)
    },
    fetchError: function (err) {
      console.error(err)
      alert("Something went wrong. Look at the logs.")
    }
  }

}
</script>
```

**Comment by markmichon** [comment]
@Slluxx I'd advise you to try and further debug or look at possibly submitting a fix if you can identify the issue. I've just done a fresh install of pizzly on heroku, created a react app based off the codesandbox linked above, and set up a discord app to test. All worked fine, and I can't reproduce the problem. If it's a vuejs conflict, that's more of an issue with how vue is being used, as pizzly-js framework agnostic.

**Comment by Slluxx** [comment]
With the help of 2 friends i figured out that 0.2.8 does not work when using insecure http. The problem here was that Pizzly on chrome thought that everything went smooth, because there was no error by the discord modal. When using safari or firefox, discord would state that the redirect url is invalid. 

I cant set up SSL in my environment so i asked the 2 friends if they would set it up independently on their machines and we discovered that it worked with any browser, just like it should.

I dont know why 0.2.5 gives me a token from heroku and then fails on CORS but on 0.2.8 it doesnt work anymore at all. I feel like it should be made very clear that you need to use https in order to use pizzly.

**Comment by paulsingh** [comment]
Chiming in here in case others are stuck with this problem. 

TLDR: the `host` param accepts the full URL. See here: 

https://github.com/Bearer/Pizzly/blob/7e2c02ae479f64b4ecca052783e038c5e0d38b7c/src/clients/javascript/src/index.ts#L69-L71

Longer explanation: 

If you've followed the example setup, you're running Pizzly on Heroku. And, by default, the Heroku URL you're likely going to use is secured by HTTPS.

The problem is that you `localhost` is likely _not_ running HTTPS. So, you'll get an error when trying to test out Pizzly from your `localhost`.

The solution is to update `host: 'pizzly.example.org'` to `host: 'https://pizzly.example.org'` for most cases.
